src/js/player.js

```diff
@@ -2076,7 +2076,15 @@ class Player extends Component {
    * @method currentSources
    */
   currentSources() {
-    return this.cache_.sources || [this.currentSource()];
+    const source = this.currentSource();
+    const sources = [];
+
+    // assume `{}` or `{ src }`
+    if (Object.keys(source).length !== 0) {
+      sources.push(source);
+    }
+
+    return this.cache_.sources || sources;
```

Comment:
Do you know off the top of your head that `sources` will never get called because `this.cache_` will always be present?

I most likely implemented this to follow suit of the original implementation—thinking that there will be a difference between `<source>` and `src`, but since the PR been open for some time I have no idea anymore.

I can investigate further if unknown.